### PR TITLE
Ts css support

### PR DESCRIPTION
**Pull request by meetdhananifynd**
- Moved webpack configuration to theme
- fixes section_meta issue
- Added setting json pull message while sync theme 
- Removed section config and custom-page config flow and merged in baseconfig
- Added Typescript support
- getAssetCdnBaseUrl Refactor

### CODE REVIEW

File: src/lib/Theme.ts [hunk 1]
```diff
@@ -1343,28 +1365,69 @@ export default class Theme {
...
+                helmetModule: {},
+                atob: globalThis.atob,
+                btoa: globalThis.btoa,
+            },
+            XMLHttpRequest: {},
+            /**
+             * console.log is an expensive synchronous logger.
+             * It is performant to patch theme implementation of console.log
+             * ! It only applies to server side evaluation
+             */
```
Comment (karanrainafynd) [inline src/lib/Theme.ts]: Remove these comments

File: src/lib/Theme.ts [hunk 2]
```diff
@@ -1343,28 +1365,69 @@ export default class Theme {
...
+                error: () => {
+                    // Overrides `console.error` methpd for theme
+                },
+                warn: () => {
+                    // Overrides `console.warn` methpd for theme
+                },
+            },
+            // Restrict `process` access to themes
+            // specifying version has a dependency on SDK
+            // It lets SDK know code is being executed in Node env
```
Comment (karanrainafynd) [inline src/lib/Theme.ts]: remove comments

File: src/lib/Theme.ts [hunk 3]
```diff
@@ -1389,6 +1452,7 @@ export default class Theme {
             const data = findExportedVariable(path, 'settings');
             try {
                 const settings = JSON.parse(data);
+                console.log(`Settings in ${path} file : `, {settings})
```
Comment (karanrainafynd) [inline src/lib/Theme.ts]: Remove

File: src/lib/Theme.ts [hunk 4]
```diff
@@ -1407,6 +1471,7 @@ export default class Theme {
             const data = findExportedVariable(path, 'sections');
             try {
                 const settings = JSON.parse(data);
+                console.log(``)
```
Comment (karanrainafynd) [inline src/lib/Theme.ts]: Remove

**Review by karanrainafynd — CHANGES_REQUESTED** [review]
Remove logs

File: src/lib/Theme.ts (hunk 2 above)
Comment (meetdhananifynd) [reply src/lib/Theme.ts]: fixed

File: src/lib/Theme.ts (hunk 3 above)
Comment (meetdhananifynd) [reply src/lib/Theme.ts]: done

File: src/lib/Theme.ts (hunk 4 above)
Comment (meetdhananifynd) [reply src/lib/Theme.ts]: done

File: src/lib/Theme.ts (hunk 2 above)
Comment (meetdhananifynd) [reply src/lib/Theme.ts]: done

File: src/helper/utils.ts [hunk 5]
```diff
@@ -296,16 +296,17 @@ export const isValidDomain = (domain) => {
...
+        presets: ['@babel/preset-react', '@babel/preset-typescript'],
     };
 
     try {
-        const result = babel.transformSync(jsxCode, options);
+        const result = babel.transformSync(code, options);
         return result?.code;
     } catch (error) {
-        console.error('Error transforming JSX to JS:', error);
+        console.error('Error transforming JSX/TSX to JS:', error);
```
Comment (karanrainafynd) [inline src/helper/utils.ts]: use Logger.error

**Review by karanrainafynd — COMMENTED** [review]
Pls update as per comments

File: src/helper/utils.ts (hunk 5 above)
Comment (karanrainafynd) [reply src/helper/utils.ts]: Fix merge conflicts and align with master

**Pull request merged**